client/src/Components/CommonComponent/Table/Table.jsx

```diff
@@ -198,15 +193,15 @@ class TableCmponent extends Component {
                     type="profile"
                   />
                   <Divider type="vertical" />
-                  <Icon onClick={event => editPopup(record.key, record, editHtml)}
+                  <Icon onClick={this.props.handleClick("customersPage","edit","editVisibility",record,record.pk_i_id)}
                     style={{
                       fontSize: "1.2rem",
                       color: "rgba(0, 0, 0, 0.65)"
                     }}
                     type="edit"
                   />
                   <Divider type="vertical" />
-                  <Icon onClick={this.props.handleClick("customersPage", "deleteVisibility", record.pk_i_id)}
+                  <Icon onClick={this.props.handleClick("customersPage","delete", "deleteVisibility",record,record.pk_i_id)}
```

Comment:
@Jamalat-shamallakh I think you have to change this line and line 196 so that you are passing Icon a function that can be called, instead of calling the function yourself like you're currently doing.
The simplest way would be to turn it into an arrow function: `onClick={() => this.props.handleClick("customersPage","delete",....`

Reply:
good idea thank you, I will do it :+1:

Reply:
Ohh, sorry.  I see now that your handleClick is a function that returns a function:
`handleClick = (value1, value2, id) => (e) => {`
So that's why your current code works, and my suggestion breaks everything 😅
Sorry, nevermind!